Packs/AzureNetworkSecurityGroups/Integrations/AzureNetworkSecurityGroups/AzureNetworkSecurityGroups_description.md

```diff
@@ -1,54 +1,85 @@
 In order to connect to the Azure Network Security Groups use one of the following methods:
 
 1. *Authorization Code Flow* (Recommended).
-2. *Device Code Flow*.
-3. *Azure Managed Identities Flow*.
+2. *Client Credentials Flow*
+3. *Device Code Flow*.
+4. *Azure Managed Identities Flow*.
+
+## Self-Deployed Azure App
+
+To use a self-configured Azure application, you need to add a new Azure App Registration in the Azure Portal.
+
+To add the registration, refer to the following [Microsoft article](https://learn.microsoft.com/en-us/defender-xdr/api-create-app-web?view=o365-worldwide) steps 1-8.
+
+### Required permissions
+
+- Azure Service Management - permission `user_impersonation` of type Delegated
+- Microsoft Graph - permission `offline_access` of type Delegated
+
+To add a permission:
+
+1. Navigate to **Home** > **App registrations**.
+2. Search for your app under 'all applications'.
+3. Click **API permissions** > **Add permission**.
+4.  Search for the specific Microsoft API and select the specific permission of type Delegated.
 
 ### Authentication Using the Authorization Code Flow (recommended)
 
-1. To use a self-configured Azure application, you need to add a new Azure App Registration in the Azure Portal. To add the registration, refer to the following [Microsoft article](https://docs.microsoft.com/en-us/microsoft-365/security/defender/api-create-app-web?view=o365-worldwide#create-an-app) steps 1-8.
```

Comment:
I think we still need this line, aren't we?

Reply:
we change this line location. its written under 'Self-Deployed Azure App' out of the scope of specific authorization because it is common for all of them